client.py

```diff
@@ -1,34 +1,68 @@
 import sys
+from typing import Any, Callable
+from config import *
 from netcode import ClientChannel, LOCALHOST
 from board import Board, Tile
 from rfid_reader import RfidReader
+from input_reader import Button, Encoder
 from display import Display
 
+button_red = Button(button_red_pin)
+# button_green = Button(button_green_pin)
+encoder = Encoder(encoder_first_pin, encoder_second_pin)
+
+
+def select_with_encoder(choices: list[Any], on_selection: Callable[[Any], None]) -> Any:
+    selected_index = 0
+    on_selection(choices[selected_index])
+    print("[INFO] Waiting for user input...")
+    while True:
+        button_red.update()
+        encoder.update()
+
+        if encoder.was_just_turned_left:
+            selected_index = (selected_index - 1) % len(choices)
```

Comment:
Wiem że to python, ale nie będzie jakiegoś dzikiego problemu z liczeniem modulo z liczby ujemnej? Pewnie nie, ale wolałem zapytać 🗡️

Reply:
W pythonie nie, ale generalnie jest to kwestia sporna i zależy od konwencji. Różne języki robią to różnie, czasami wynik czegoś takiego jest dodatni, czasami ujemny, a czasami masz oddzielne operacje na oba przypadki. No różne są